### PR TITLE
Expose APIs

### PR DESCRIPTION
**Pull request by b1llow**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

File: librz/include/rz_core.h [hunk 1]
```diff
@@ -418,6 +418,20 @@ RZ_API void rz_core_item_free(RzCoreItem *ci);
...
+	char *rz_symbol_name;
+	char *libname; // name of the lib this symbol is specific to, if any
+	char *nameflag; // flag name for symbol
+	char *demname; // demangled raw symbol name
+	char *demflag; // flag name for demangled symbol
+	char *classname; // classname
+	char *classflag; // flag for classname
+	char *methname; // methods [class]::[method]
+	char *methflag; // methods flag sym.[class].[method]
+} SymName;
```
Comment (ret2libc) [inline librz/include/rz_core.h]: This should have a doxygen description, moreover any exported structure should start with `Rz`.

File: librz/include/rz_core.h [hunk 2]
```diff
@@ -457,6 +471,7 @@ RZ_API int rz_core_config_init(RzCore *core);
 RZ_API bool rz_core_config_eval_and_print(RzCore *core, const char *str, bool many);
 RZ_API void rz_core_config_print_all(RzConfig *cfg, const char *str, RzCmdStateOutput *state);
 RZ_API void rz_core_parse_rizinrc(RzCore *r);
+RZ_API RZ_OWN RzList *rz_core_config_variable_spaces(RZ_NONNULL RzCore *core, RZ_NULLABLE const char *space);
```
Comment (ret2libc) [inline librz/include/rz_core.h]: Shall this be named `rz_core_config_in_space`?

Comment (b1llow) [reply librz/include/rz_core.h]: I don't know how to come up with a proper name

Comment (wargio) [reply librz/include/rz_core.h]: i like ret2libc proposal.

File: librz/include/rz_core.h [hunk 3]
```diff
@@ -418,19 +418,24 @@ RZ_API void rz_core_item_free(RzCoreItem *ci);
...
+	char *symbolname; ///< display symbol name
+	char *libname; ///< name of the lib this symbol is specific to, if any
+	char *nameflag; ///< flag name for symbol
+	char *demname; ///< demangled raw symbol name
+	char *demflag; ///< flag name for demangled symbol
+	char *classname; ///< classname
+	char *classflag; ///< flag for classname
+	char *methname; ///< methods [class]::[method]
+	char *methflag; ///< methods flag sym.[class].[method]
+} RzSymName;
```
Comment (ret2libc) [inline librz/include/rz_core.h]: @XVilka ? WDYT? Better name maybe? It has a very specific use case but the name is very generic, I'm not 100% sure it is good as-is.

File: librz/core/cbin.c [hunk 4]
```diff
@@ -1659,10 +1664,8 @@ RZ_API RZ_OWN HtPP *rz_core_bin_create_digests(RzCore *core, ut64 paddr, ut64 si
 	return r;
 }
 
-RZ_API int rz_core_bin_set_cur(RzCore *core, RzBinFile *binfile) {
-	if (!core->bin) {
-		return false;
-	}
+RZ_API bool rz_core_bin_set_cur(RZ_NONNULL RzCore *core, RZ_NULLABLE RzBinFile *binfile) {
```
Comment (XVilka) [inline librz/core/cbin.c:1673]: Missing Doxygen.

File: librz/core/cbin.c [hunk 5]
```diff
@@ -1692,7 +1695,12 @@ static RZ_NULLABLE RZ_BORROW const RzList *core_bin_strings(RzCore *r, RzBinFile
 	return rz_bin_get_strings(r->bin);
 }
 
-static const char *get_compile_time(Sdb *binFileSdb) {
+/**
+ * \brief Get TimeDateStamp string
```
Comment (XVilka) [inline librz/core/cbin.c]: I think this description is hard to understand for a newbie

File: librz/core/cbin.c [hunk 6]
```diff
@@ -2854,13 +2862,20 @@ RZ_API bool rz_core_bin_whole_strings_print(RzCore *core, RzBinFile *bf, RzCmdSt
...
 		free(bf->file);
 		free(bf);
 	}
+	return l;
+}
+
+RZ_API bool rz_core_bin_whole_strings_print(RzCore *core, RzBinFile *bf, RzCmdStateOutput *state) {
+	rz_return_val_if_fail(core && state, false);
+
+	RzList *l = rz_core_bin_whole_strings(core, bf);
```
Comment (XVilka) [inline librz/core/cbin.c:2882]: It can return NULL, probably worth doing a quick return in this case

File: librz/core/ctypes.c [hunk 7]
```diff
@@ -415,8 +415,14 @@ RZ_IPI RZ_OWN char *rz_core_base_type_as_c(RzCore *core, RZ_NONNULL RzBaseType *
...
+ * \brief Get a type string by \p name (\see rz_type_db_base_type_as_pretty_string)
+ * \param core RzCore reference
+ * \param name Type name
+ * \param multiline Pretty printing with RZ_TYPE_PRINT_MULTILINE
+ */
+RZ_API RZ_OWN char *rz_core_types_as_c(RZ_NONNULL RzCore *core, RZ_NONNULL const char *name, bool multiline) {
+	rz_return_val_if_fail(core && core->analysis && name, NULL);
 
 	RzBaseType *btype = rz_type_db_get_base_type(core->analysis->typedb, name);
 	if (!btype) {
```
Comment (XVilka) [inline librz/core/ctypes.c:428]: return NULL below, while you are on it.

File: librz/include/rz_core.h (hunk 3 above)
Comment (XVilka) [reply librz/include/rz_core.h]: I think, `RzBinSymNames` would be a better choice, note the *Names*

**Review by XVilka — CHANGES_REQUESTED** [review]
Looks great! Only a few small nitpicks and good to merge 👍

**Review by XVilka — APPROVED** [review]

**Comment by XVilka** [comment]
@imbillow please rebase to resolve conflicts, maybe squash too.

File: librz/include/rz_core.h (hunk 3 above)
Comment (ret2libc) [reply librz/include/rz_core.h]: RzCoreSymNames. It's in RzCore, not RzBin.

**Comment by XVilka** [comment]
@imbillow once you do that and it passes CI - feel free to merge it yourself, just be sure to squash and have a good commit message.

**Pull request merged**